libbeat/docs/generalconfig.asciidoc

```diff
@@ -103,3 +103,70 @@ processors in your config.
 
 Sets the maximum number of CPUs that can be executing simultaneously. The
 default is the number of logical CPUs available in the system.
+
+[float]
+[[configuration-internal-queue]]
+=== Configure the internal queue
+
+The `queue` namespace configures the queue type to be used.
+
+The Elastic Beats employ an internal queue for events to be published. The
+queue is responsible for buffering and combining events into batches, which can
+be consumed by the outputs. The outputs will use bulk operations to send a
+batch of events in one transaction.
+
+
+Example configuration:
+
+[source,yaml]
+------------------------------------------------------------------------------
+queue.mem:
+  events: 4096
+------------------------------------------------------------------------------
+
+==== Configure the Memory Qeueue
+[[configuration-internal-queue-memory]]
+
+The memory queue keeps all events in memory. By default no flush interval is
+configured. All events published to this queue will be directly consumed by the
+outputs. An output will consume up to the outputs `bulk_max_size` events at once.
```

Comment:
I like when we can be concise, but not if we're obfuscating meaning. In this case, I think the meaning is not clear and the grammatical structure is questionable because you are leaving words out. What you're really saying here is that "An output will consume up to the number of events specified by the output's `bulk_max_size` setting." You're using shorthand that's common for developers to use, but will probably seem off to some readers. They won't make the same substitution that you make in your head. 

Anyhow, for now, at least make "output" possessive so that you have "the output's `bulk_max_size`.